tests/ui/closure-expected-type/expect-infer-var-appearing-twice.stderr

```diff
@@ -9,8 +9,8 @@ LL | |
 LL | |     });
    | |______^ expected due to this
    |
-   = note: expected closure signature `fn(_, _) -> _`
-              found closure signature `fn(u32, i32) -> _`
+   = note: expected closure signature `fn(_, u32) -> _`
+              found closure signature `fn(_, i32) -> _`
```

Comment:
not caused by the lhs now being an infer var, but instead by `TypeErrCtxt::cmp` eliding it because the types are the same. I have to admit that I really dislike that function